### PR TITLE
Paywalls for User invites, sharing, and group management

### PR DESCRIPTION
**Pull request by MrFlashAccount**
#### Tl;dr 
This PR adds paywalls for Invites, Sharing, Members settings, and User Group settings 

<details><summary>Demo Presentation</summary>
<p>

### Note: By the time of recording, the User groups page didn't feel well due to regression.

https://github.com/enso-org/enso/assets/61194245/cd7c0579-70cb-4b12-885f-5791624eb38f

</p>
</details> 

---

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Comment by PabloBuchu** [comment]
Even more ts errors.. Am I doing something wrong?

```
 INFO ide_ci::program::command: npmℹ️ src/components/dashboard/column/SharedWithColumn.tsx(6,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/components/dashboard/column/SharedWithColumn.tsx(18,26): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/components/dashboard/columnHeading/SharedWithColumnHeading.tsx(6,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/components/dashboard/columnHeading/SharedWithColumnHeading.tsx(15,26): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/AssetContextMenu.tsx(6,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/AssetContextMenu.tsx(26,26): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTab.tsx(6,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTab.tsx(37,33): error TS2551: Property 'useFullUserSession' does not exist on type 'typeof import("/Users/pawelbuchowski/projects/enso/app/ide-desktop/lib/dashboard/src/providers/AuthProvider")'. Did you mean 'useUserSession'?
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTabBar.tsx(4,36): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTabBar.tsx(9,36): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTabBar.tsx(45,39): error TS2345: Argument of type '"noSeatsLeft"' is not assignable to parameter of type '"type" | "status" | "value" | "settings" | "name" | "description" | "color" | "delete" | "share" | "copy" | "uploadFiles" | "newProject" | "newFolder" | "newSecret" | "newDatalink" | ... 576 more ... | "enterprisePlan"'.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/MembersSettingsTabBar.tsx(45,64): error TS2345: Argument of type '"seatsLeft"' is not assignable to parameter of type '"type" | "status" | "value" | "settings" | "name" | "description" | "color" | "delete" | "share" | "copy" | "uploadFiles" | "newProject" | "newFolder" | "newSecret" | "newDatalink" | ... 576 more ... | "enterprisePlan"'.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/withPaywall.tsx(13,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/withPaywall.tsx(17,36): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/Settings/withPaywall.tsx(57,35): error TS2551: Property 'useFullUserSession' does not exist on type 'typeof import("/Users/pawelbuchowski/projects/enso/app/ide-desktop/lib/dashboard/src/providers/AuthProvider")'. Did you mean 'useUserSession'?
 INFO ide_ci::program::command: npmℹ️ src/layouts/UserBar.tsx(9,26): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/layouts/UserBar.tsx(21,26): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/modals/InviteUsersModal/InviteUsersForm.tsx(7,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/modals/InviteUsersModal/InviteUsersForm.tsx(16,36): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/modals/InviteUsersModal/InviteUsersForm.tsx(45,33): error TS2551: Property 'useFullUserSession' does not exist on type 'typeof import("/Users/pawelbuchowski/projects/enso/app/ide-desktop/lib/dashboard/src/providers/AuthProvider")'. Did you mean 'useUserSession'?
 INFO ide_ci::program::command: npmℹ️ src/modals/InviteUsersModal/InviteUsersForm.tsx(133,24): error TS2345: Argument of type '"inviteFormSeatsLeftError"' is not assignable to parameter of type '"type" | "status" | "value" | "settings" | "name" | "description" | "color" | "delete" | "share" | "copy" | "uploadFiles" | "newProject" | "newFolder" | "newSecret" | "newDatalink" | ... 576 more ... | "enterprisePlan"'.
 INFO ide_ci::program::command: npmℹ️ src/modals/InviteUsersModal/InviteUsersForm.tsx(222,26): error TS2345: Argument of type '"inviteFormSeatsLeft"' is not assignable to parameter of type '"type" | "status" | "value" | "settings" | "name" | "description" | "color" | "delete" | "share" | "copy" | "uploadFiles" | "newProject" | "newFolder" | "newSecret" | "newDatalink" | ... 576 more ... | "enterprisePlan"'.
 INFO ide_ci::program::command: npmℹ️ src/modals/ManagePermissionsModal.tsx(8,31): error TS2307: Cannot find module '#/hooks/billing' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/modals/ManagePermissionsModal.tsx(22,26): error TS2307: Cannot find module '#/components/Paywall' or its corresponding type declarations.
 INFO ide_ci::program::command: npmℹ️ src/modals/ManagePermissionsModal.tsx(64,33): error TS2551: Property 'useFullUserSession' does not exist on type 'typeof import("/Users/pawelbuchowski/projects/enso/app/ide-desktop/lib/dashboard/src/providers/AuthProvider")'. Did you mean 'useUserSession'?
 INFO ide_ci::program::command: npmℹ️ src/modals/ManagePermissionsModal.tsx(409,70): error TS2345: Argument of type '"shareFullPaywallMessage"' is not assignable to parameter of type '"type" | "status" | "value" | "settings" | "name" | "description" | "color" | "delete" | "share" | "copy" | "uploadFiles" | "newProject" | "newFolder" | "newSecret" | "newDatalink" | ... 576 more ... | "enterprisePlan"'.
 ```

**Comment by MrFlashAccount** [comment]
@PabloBuchu A little bit yes 😅 
CleanShot 2024-06-10 at 15 13 35

**Comment by PabloBuchu** [comment]
@MrFlashAccount but still I cant test https://github.com/enso-org/enso/pull/10202 cause it has typescript errors while I am building Enso.

**Comment by PabloBuchu** [comment]
Hmm so why this PR is `ready for review` if I can't even qa it? If this points to `develop` then I assume I should be able to at least build it and run a qa, no? If you depend on other PRs then just create chain / linked PRs

**Comment by MrFlashAccount** [comment]
> @MrFlashAccount but still I cant test #10202 cause it has typescript errors while I am building Enso.

That PR(#10202) wasn't supposed to be tested, since it only adds components that are not in use yet, we can skip QA for that PR and do QA in this PR once it's ready for review/QA (once we merge dependencies)

This approach isn't perfect, I think I should've left this PR in the `draft` state until deps are merged. And currently, we can't do QA for components that are not present somewhere in the UI. It's a drawback of the current process and we'll fix it eventually with the Storybook. (We'll use it as a playground and testing platform for components)

If you have ideas on how we can improve the process it the future, feel free to provide suggestions.

**Comment by PabloBuchu** [comment]
1. Contact Sales for Unlimited User Groups should have the same blue color like the rest
Screenshot 2024-06-18 at 14 09 11

2. Can we make the padlock to be gold? Everywhere both in buttons and in text. Will be better visible
3. The one next to shared with looks like not centerd
Screenshot 2024-06-18 at 14 10 53

Very good job with `Devtools`!

**Comment by MrFlashAccount** [comment]
> Contact Sales for Unlimited User Groups should have the same blue color like the rest

I decided to use "primary" color here to match the button on subscribe page. Also it makes sense to separate these buttons visually since they lead to different actions (upgrade vs write an email)

**Comment by MrFlashAccount** [comment]
> Can we make the padlock to be gold? Everywhere both in buttons and in text. Will be better visible

Gold usually has less contrast ratio than black. We can make it large to make it more visible. Icon in buttons matches the current design

**Comment by MrFlashAccount** [comment]
> Very good job with Devtools!

Thanks! We're going to add there more product-specific features in the future 🙂

**Comment by MrFlashAccount** [comment]
@PabloBuchu alighned button with the text. Thanks for noticing!

**Comment by PabloBuchu** [comment]
> I decided to use "primary" color here to match the button on subscribe page. Also it makes sense to separate these buttons visually since they lead to different actions (upgrade vs write an email)

But this is the same from user perspective but different way :) even its just `mailto:` it leads to the same result upgrading Enso to Enterprise

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
QA ✅ We can fix minor things later after more general review
